relations/src/hasher.rs

```diff
@@ -0,0 +1,83 @@
+use halo2_base::{
+    gates::GateChip, poseidon::hasher::PoseidonHasher, utils::BigPrimeField, AssignedValue, Context,
+};
+
+use crate::{account::CircuitAccount, note::CircuitNote};
+
+pub trait InnerHasher<F: BigPrimeField> {
+    fn hash_account(
+        &mut self,
+        ctx: &mut Context<F>,
+        gate: &GateChip<F>,
+        account: &impl CircuitAccount<F>,
+    ) -> AssignedValue<F>;
+
+    fn hash_note(
+        &mut self,
+        ctx: &mut Context<F>,
+        gate: &GateChip<F>,
+        note: &CircuitNote<F>,
+    ) -> AssignedValue<F>;
```

Comment:
I think that the hasher doesn't have to know what it is hashing - it'd be sufficient to have a single method that hashes anything that can be turned into an array

Reply:
Same as above